### PR TITLE
Add view command

### LINKED ISSUE #104: View command with UI change
Adding new commands to split screen and show person card
- view <INDEX> will result in screen splitting (planning of 50:50 or 60:40 of list:card) and show details for a person

### PR DESCRIPTION
**Pull request by iamdiluxedbutcooler**
closes #104 

Screenshot 2567-10-22 at 05 18 28

 This command allows users to see detailed information about a specific person in the address book using the following syntax: `view <INDEX>`

### CODE REVIEW

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 1]
```diff
@@ -65,20 +65,16 @@ private void createFields() {
...
-        }
+        // Apply the existing style classes
+        tierLabel.getStyleClass().add("label");
+
+        // Add the tier-specific style class
+        String tier = person.getTier().toParsableString().toLowerCase();
+        tierLabel.getStyleClass().add(tier + "-tier");
+
+        // Add the label to the FlowPane
+        assignedTier.getChildren().add(tierLabel);
```
Comment (FionaQY) [inline src/main/java/seedu/address/ui/PersonCard.java:78]: Good idea. the issue now is that for those with no tier, there is now an awkward gap where Tier originally was.

**Review by FionaQY — APPROVED** [review]
LGTM. Good job! The Ui looks pretty good

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -111,6 +118,43 @@ public void setPerson(Person target, Person editedPerson) {
...
+        }
+        selectedPerson.setValue(person);
+    }
+
+    /**
+     * Ensures {@code selectedPerson} is a valid person in {@code filteredPersons}.
+     */
+    private void ensureSelectedPersonIsValid(ListChangeListener.Change<? extends Person> change) {
+        while (change.next()) {
+            if (selectedPerson.getValue() == null) {
```
Comment (ZShunRen) [inline src/main/java/seedu/address/model/ModelManager.java:146]: perhaps you can use `getSelectedPerson` method here to improve maintainability.

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -163,8 +180,33 @@ private void handleExit() {
...
+                int index = Integer.parseInt(commandParts[1]) - 1; // Assuming 1-based indexing in UI
+                Person personToView = logic.getFilteredPersonList().get(index);
+                personDetailPanel.setPersonDetails(personToView);
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                // Handle invalid index
+                resultDisplay.setFeedbackToUser("Invalid person index.");
+            }
+        } else {
+            resultDisplay.setFeedbackToUser("Please provide a person index to view.");
+        }
```
Comment (ZShunRen) [inline src/main/java/seedu/address/ui/MainWindow.java:209]: nicely done!

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 1 above)
Comment (ZShunRen) [reply src/main/java/seedu/address/ui/PersonCard.java:78]: To add on, I think you can use a conditional to render and add the label only if the `tier != "NA"`

**Review by ZShunRen — APPROVED** [review]
Terrific work pleng! I really like the new UI changes with view. Although close command has yet to be added, the tests you have added are also very comprehensive!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -111,6 +118,43 @@ public void setPerson(Person target, Person editedPerson) {
...
+    @Override
+    public Person getSelectedPerson() {
+        return selectedPerson.getValue();
+    }
+
+    @Override
+    public void setSelectedPerson(Person person) {
+        if (person != null && !filteredPersons.contains(person)) {
+            throw new PersonNotFoundException();
+        }
```
Comment (itsme-zeix) [inline src/main/java/seedu/address/model/ModelManager.java:137]:
```suggestion
        requireNonNull(person);
        if (!filteredPersons.contains(person)) {
            throw new PersonNotFoundException();
        }
```
Could consider doing this to match what the module did, but I have no issues with the current implementation either.

File: src/main/java/seedu/address/model/ModelManager.java (hunk 2 above)
Comment (itsme-zeix) [inline src/main/java/seedu/address/model/ModelManager.java:146]:
```suggestion
            if (this.getSelectedPerson() == null) {
```

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -111,6 +118,43 @@ public void setPerson(Person target, Person editedPerson) {
...
+    private void ensureSelectedPersonIsValid(ListChangeListener.Change<? extends Person> change) {
+        while (change.next()) {
+            if (selectedPerson.getValue() == null) {
+                return;
+            }
+
+            boolean wasSelectedPersonRemoved = change.getRemoved().stream()
+                    .anyMatch(removedPerson -> selectedPerson.getValue().isSamePerson(removedPerson));
+            if (wasSelectedPersonRemoved) {
+                selectedPerson.setValue(change.getFrom() > 0 ? filteredPersons.get(change.getFrom() - 1) : null);
```
Comment (itsme-zeix) [inline src/main/java/seedu/address/model/ModelManager.java:153]:
```suggestion
                this.setSelectedPerson(change.getFrom() > 0 ? filteredPersons.get(change.getFrom() - 1) : null);
```

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 6]
```diff
@@ -175,10 +217,13 @@ public PersonListPanel getPersonListPanel() {
     private CommandResult executeCommand(String commandText) throws CommandException, ParseException {
         try {
             CommandResult commandResult = logic.execute(commandText);
-            logger.info("Result: " + commandResult.getFeedbackToUser()); // used to log the result, not actually
-            // used to display
+            logger.info("Result: " + commandResult.getFeedbackToUser());
             resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());
 
+            if (commandText.trim().toLowerCase().startsWith("view")) {
```
Comment (itsme-zeix) [inline src/main/java/seedu/address/ui/MainWindow.java:223]: Might be worth refactoring the predicate condition to a `isViewCommand()` method, similar to the `isShowHelp()` method.

**Review by itsme-zeix — APPROVED** [review]
Awesome work - super clean code and looks great!

We should discuss whether a user image is necessary since there is no function to change it at the moment but we can just merge anyways, we can deal with it later.

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 1 above)
Comment (iamdiluxedbutcooler) [reply src/main/java/seedu/address/ui/PersonCard.java:78]: Thanks for feedback guys @FionaQY @ZShunRen . Have opened a new issue and picked up: #119

File: src/main/java/seedu/address/model/ModelManager.java (hunk 4 above)
Comment (iamdiluxedbutcooler) [reply src/main/java/seedu/address/model/ModelManager.java:137]: Thanks for pointing out!

**Pull request merged**